brainiak/searchlight/searchlight.py

```diff
@@ -0,0 +1,203 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+"""Distributed searchlight
+Given a function and a list of volumes, this applies the function
+to all voxels. 
+"""
+
+# Author: Michael Anderson
+
+__all__ = [
+    "Searchlight",
+]
+
+
+from mpi4py import MPI
+import numpy as np
+
+class Searchlight:
```

Comment:
Actually it could be thought as a transformer. you get input and you transform the input to the specific analysis. Examples you can look at in sklearn are the dimensionality reduction methods such as PCA, ICA, etc...
Maybe we should also check nilearn.